retrospective-update/hydrosos_maps.py

```
@@ -0,0 +1,270 @@
+            flow_data = (
+                monthly_ds
+                .sel(time=date_str)
+                ["Q"]
+                .mean(dim="time", skipna=True)
+            )
+
+            # Compute duplicates
+            results = []
+            for hybas_id, linknos in hybas_groups.items():
+                try:
+                    linknos = [int(l) for l in linknos if pd.notna(l)]
```

Ruff (0.12.2)

Ambiguous variable name: `l`

(E741)